SVPAlib/src/automata/sra/SRACheckMove.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * SVPAlib
+ * automata
+ * Jul 25, 2018
+ * @author Tiago Ferreira
+ */
+package automata.sra;
+
+import org.sat4j.specs.TimeoutException;
+
+import theory.BooleanAlgebra;
+
+/**
+ * SRACheckMove
+ * @param <P> set of predicates over the domain S
+ * @param <S> domain of the automaton alphabet
+ */
+public class SRACheckMove<P, S> extends SRAMove<P, S> {
+
+	/**
+	 * Constructs an SRA Check Transition that starts from state <code>from</code> and ends at state
+	 * <code>to</code> with input <code>input</code>
+     * Check transitions happen iff the predicate is true and the register mentioned matches the input symbol.
+	 */
+	public SRACheckMove(Integer from, Integer to, P guard, Integer register) {
+		super(from, to, guard, register);
+    }	
+
+	@Override
+	public boolean isSatisfiable(BooleanAlgebra<P, S> boolal) throws TimeoutException {
+        // FIXME: Requires knowledge of the registers at time of transition.
```

Comment:
For satisfiability we have to check if not only the `guard` is satisfiable, but also if having the input already in the predicate of the register is satisfiable. Right now we can't do this as the move has no knowledge of the actual content of the registers, just which one it is checking.

Reply:
For now, you can just check that the move guard is satisfiable. We'll deal with more complex questions later

Reply:
Sounds good